Makefile

```
@@ -201,12 +201,29 @@ xgo:
 install-os-dependencies:
 	_assets/scripts/install_deps.sh
 
-setup: install-os-dependencies dep-install lint-install mock-install gen-install update-fleet-config ##@other Prepare project for first build
+setup: install-os-dependencies dep-install lint-install mock-install deploy-install gen-install update-fleet-config ##@other Prepare project for first build
 
 generate: ##@other Regenerate assets and other auto-generated stuff
 	go generate ./static ./static/migrations
 	$(shell cd ./services/shhext/chat && exec protoc --go_out=. ./*.proto)
 
+prepare-release:
```

Should we just out `clean-release` as a requirement?
```suggestion
prepare-release: clean-release
```

---

> The alternative to this process might be relying solely on a value in VERSION. What I mean by that is when make deploy is run, we look at a value in VERSION and this is our tag. A git tag and GH release are created and then VERSION is bumped and committed to develop. What is not deterministic is the bump. We can default to patch and if it turns out a breaking change is introduced, a committer of that change would be responsible to bump minor.

This would be my preferred workflow, I use that in a few projects, basically once we want to release, we change the `VERSION`, bumping by a minor/major depending on breaking changes, and then run make deploy, which solely looks at `VERSION`. We can run some safety check to make sure there is not already a release with that version (someone forgot to bump before releasing), allowing possibly to override (in case of broken releases etc).

I would not automate the bumping of `VERSION` (i.e. having the CI system do that), as I prefer this action to be explicit (we can of course provide a script to bump minor/major).

---

> I would not automate the bumping of VERSION (i.e. having the CI system do that), as I prefer this action to be explicit (we can of course provide a script to bump minor/major).

The problem here is that after a release we will keep the old version and if someone creates alpha, beta etc., it will refer as `already.released.version-beta` while it should be `upcoming.release-beta`.

I agree with the rest. Let's create `VERSION` file then and simplify the workflow.